### PR TITLE
FEAT(): Added command interpreting and WorkoutManager

### PR DESCRIPTION
**Pull request by timothyheadrick12**
Despite the branch name, this should not impact original functionality. It adds a board.h file with "DEBUG 1" defined, change it to "DEBUG 0" to get original functionality. 

WorkoutManager handles incoming commands over the UART ble service and handles whether IMU data is being read or not.

Since this does not change the original functionality of the firmware, it will only properly read "IMU data" when in debug mode, in which case dummy data is supplied. Another update would be necessary to add real IMU data transmission to the app, but I am waiting to discuss first.

### CODE REVIEW

**Comment by demcdonald2000** [comment]
@timothyheadrick12 I reviewed the changes from my phone but haven't tested them on my board. I've left a few comments, but otherwise I think I'm following and that it looks good. 

Something to note: Serial.println() takes somewhere around 10-20ms to execute, I found that using Serial.write() is much faster by just printing a raw byte string. You'll have to add newlines and maybe used a different type of string, but overall it has a very positive impact on the execution time of the code. Since we won't be using actual IMU data it shouldn't be a problem, but as soon as we take it out of debug mode it may interfere with the IMU timing. You may know this already and have used println() for a particular reason, but it wasn't immediately obvious to me.

File: src/BluetoothManager.cpp [hunk 1]
```diff
@@ -179,6 +186,48 @@ void set_imu_characteristics(BNO08x_RVC_Data *heading)
...
+    // // Serial.write("Pitch updated.\n");
+    // pRollCharacteristic->setValue(DEFAULT_ROLL);
+    // // Serial.write("Roll updated.\n");
+    // pXAccelCharacteristic->setValue(DEFAULT_X_ACCEL);
+    // // Serial.write("X accel updated.\n");
+    // pYAccelCharacteristic->setValue(DEFAULT_Y_ACCEL);
+    // Serial.write("Y accel updated.\n");
+    pZAccelCharacteristic->setValue(data, 3 * sizeof(float) + sizeof(uint64_t));
+    pZAccelCharacteristic->notify();
+    // Serial.write("Z accel updated.\n");
```
Comment (demcdonald2000) [inline src/BluetoothManager.cpp:211]: We'll need to update these to remove unnecessary characteristics at some point if we are to use a single characteristic for the acceleration data and the timer. We should also add one to hold the yaw, pitch, and roll I think.

Comment (timothyheadrick12) [reply src/BluetoothManager.cpp:211]: That's what I was waiting to talk about during the meeting, but I could also do it today while I work on some other things.

Comment (demcdonald2000) [reply src/BluetoothManager.cpp:211]: I think I'm on the same page as you, no need to wait in my mind if you have the time.  Check my understanding. You would be:
- removing all commented out lines
- updating the name of the ZAccelCharacteristic variable
- adding a characteristic to hold the yaw, pitch, and roll

**Pull request merged**